docs/index.mdx

```diff
@@ -1,17 +1,31 @@
 ---
+title: Docs Home
```

Comment:
In the Helm 2 & 3 docs, this top level `index.mdx` page is called Docs Home, and includes the basic docs "welcome" statement as well as the docs cards component.

To preserve this welcome/Docs Home page for the first iteration of Helm 4 docs, I did the following:
- Copied over the basic Welcome statement from v3
- Moved the Helm 4-specific content over to Helm 4 Overview (`overview.md`)